src/main.ts

```diff
@@ -140,6 +141,13 @@ async function main() {
 				if (response) {
 					message.reply(response);
 				}
+
+				const spamPrevention = await messageIsSpam(message);
+				if(spamPrevention){
+					message.delete();
+					message.member?.timeout(config.options.timeoutTime, config.messages.timeoutSpamReason);
+					(client.channels.cache.get(config.channels.modChannel) as TextChannel).send("User <@"+message.author.id+"> Has send more than "+config.options.maxMentions+" mentions and therefore has been timed out for "+config.options.timeoutTime+" milliseconds.");
```

Comment:
Could we probably add this as a message like in log.ts like we do with message deletions, bans and updates? I feel like doing it there would be better since those types of messages automatically send it to our log channel.

<https://github.com/ChaosInitiative/p2ce-discord-bot/blob/main/src/utils/log.ts#L60>
<https://github.com/ChaosInitiative/p2ce-discord-bot/blob/main/src/main.ts#L130>

Reply:
The problem is I don't know how you'd get UID and the other things if it's a from the config message. I had trouble thinking about this exact thing but didn't really know how to resolve it without bloating what I wanted to be a small commit.

Reply:
Well you have access to the message since you delete it and time out the author, that is effectively what is done with messageDeleted so I don't see why that sort of method couldn't be made here.